### ISSUE TITLE
Add support to Windows XP

### ISSUE THREAD

**Issue by hainguyents13** [hainguyents13/mechvibes#4]
**@hainguyents13**
there is no error detail,  just a little window show up and says ''W7 or later are needed''
with other apps when they don't support XP you get the typical error window ''it's not a valid Win32 application''
with other apps you can also see different errors on startup but they end up working fine.

PC specs ? ASUS mainboard(s) and AMD or intel quadcore 
(XP is only an issue to find USB or graphic drivers on very recent hardware)

When a tool doesn't run on XP I test it on another PC with W7  to see wether it's worth keeping it.
I found your utility really nice to me this is why I'm trying to convince you to backport it to XP.

I found out your app in this site from India
https://www.thewindowsclub.com/mechvibes-play-mechanical-key-press-sound

_Originally posted by @andika207 in https://github.com/hainguyents13/mechvibes/issues/1#issuecomment-576052066_

**Comment by hainguyents13** [comment]
@andika207 sorry for closing this issue without having the problem fixed, after many tries I still cannot get Nodejs run on Windows XP. truely sorry!

**Comment by andika207** [comment]
@hainguyents13  do you use XP ? how do you know it doesn't run well ?
if I was any good on coding I would try myself to back port it to XP 
very sad news because I have not found any similar utility :(

**Comment by hainguyents13** [comment]
Hi, I dont have any machine running windows xp so I installed it on a virtual machine (virtual box) but nodejs said it only supports windows 7 or newer. 
So if you want to back port it yourself you can try to download and install nodejs version 10 or 8 (https://nodejs.org/en/download/releases/)
After getting nodejs running successfully, please let me know!
For any further information, please mail me at hainguyen.ts13@gmail.com for faster communication.

**Comment by hainguyents13** [comment]
Just mention in case you miss it @andika207

**Comment by andika207** [comment]
@hainguyents13  the last release compatible with XP is **6.12.3** thanks

**Comment by hainguyents13** [comment]
> @hainguyents13 the last release compatible with XP is **6.12.3** thanks

Hey, so you get nodejs worked? 
- Now download https://github.com/hainguyents13/mechvibes/archive/v2.0.1.zip then extract it,
- Open command promt in extracted folder
- Run ``npm install``
- If there is no error, run ``npm start`` and see if it work, then let me know

**Comment by andika207** [comment]
sorry for the confusion, the 6.12.3 does not support XP.
I had only downloaded the portable version (zip file) 
that version did not show the typical error message = ''it's not a valid win32 app'' 

Then I have selected an older random version 4.1.2 (msi file) and got it working
now it's up to you to sort this bug out. thanks
https://i.postimg.cc/ncyRXdw9/screenshot-18.png

**Comment by andika207** [comment]
sorry for the confusion, the 6.12.3 does not support XP.
I had only downloaded the portable version (zip file)
that version did not show the typical error message = ''it's not a valid win32 app''

Then I have selected an older random version 4.1.2 (msi file) and got it working
now it's up to you to sort this bug out. thanks
https://i.postimg.cc/QhtRnwQm/screenshot-18.png

**Comment by andika207** [comment]
**npm-debug**

0 info it worked if it ends with ok
1 verbose cli [ 'K:\\Archivos de programa\\nodejs\\node.exe',
1 verbose cli   'K:\\Archivos de programa\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'start' ]
2 info using npm@2.14.4
3 info using node@v4.1.2
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info prestart mechvibes@2.0.1
6 info start mechvibes@2.0.1
7 verbose unsafe-perm in lifecycle true
8 info mechvibes@2.0.1 Failed to exec start script
9 verbose stack Error: mechvibes@2.0.1 start: `electron .`
9 verbose stack Exit status 3221225785
9 verbose stack     at EventEmitter.<anonymous> (K:\Archivos de programa\nodejs\node_modules\npm\lib\utils\lifecycle.js:214:16)
9 verbose stack     at emitTwo (events.js:87:13)
9 verbose stack     at EventEmitter.emit (events.js:172:7)
9 verbose stack     at ChildProcess.<anonymous> (K:\Archivos de programa\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
9 verbose stack     at emitTwo (events.js:87:13)
9 verbose stack     at ChildProcess.emit (events.js:172:7)
9 verbose stack     at maybeClose (internal/child_process.js:818:16)
9 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
10 verbose pkgid mechvibes@2.0.1
11 verbose cwd K:\Documents and Settings\SINAL\Escritorio\mech\mech1
12 error Windows_NT 5.1.2600
13 error argv "K:\\Archivos de programa\\nodejs\\node.exe" "K:\\Archivos de programa\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
14 error node v4.1.2
15 error npm  v2.14.4
16 error code ELIFECYCLE
17 error mechvibes@2.0.1 start: `electron .`
17 error Exit status 3221225785
18 error Failed at the mechvibes@2.0.1 start script 'electron .'.
18 error This is most likely a problem with the mechvibes package,
18 error not with npm itself.
18 error Tell the author that this fails on your system:
18 error     electron .
18 error You can get their info via:
18 error     npm owner ls mechvibes
18 error There is likely additional logging output above.
19 verbose exit [ 1, true ]

**Comment by hainguyents13** [comment]
can you run ``npm install -g electron`` then run ``npm start`` again?

**Comment by andika207** [comment]
**npm-debug** (don't know why some lines are crossed out)

0 info it worked if it ends with ok
1 verbose cli [ 'K:\\Archivos de programa\\nodejs\\node.exe',
1 verbose cli   'K:\\Archivos de programa\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'start' ]
2 info using npm@3.7.3
3 info using node@v5.9.1
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle mechvibes@2.0.1~prestart: mechvibes@2.0.1
6 silly lifecycle mechvibes@2.0.1~prestart: no script for prestart, continuing
7 info lifecycle mechvibes@2.0.1~start: mechvibes@2.0.1
8 verbose lifecycle mechvibes@2.0.1~start: unsafe-perm in lifecycle true
9 verbose lifecycle mechvibes@2.0.1~start: PATH: K:\Archivos de programa\nodejs\node_modules\npm\bin\node-gyp-bin;K:\Documents and Settings\SINAL\Escritorio\mech\mech1\node_modules\.bin;K:\Archivos de programa\nodejs;K:\Documents and Settings\SINAL\Datos de programa\npm;K:\Archivos de programa\nodejs\;K:\WINDOWS\system32;K:\WINDOWS;K:\WINDOWS\System32\Wbem;K:\Archivos de programa\Universal Extractor;K:\Archivos de programa\Universal Extractor\bin;k:\archivos de programa\d3dgear;k:\archivos de programa\d3dgear;K:\Archivos de programa\Archivos comunes\Seagate\SnapAPI\;K:\Archivos de programa\nodejs\;K:\Documents and Settings\SINAL\Datos de programa\npm
10 verbose lifecycle mechvibes@2.0.1~start: CWD: K:\Documents and Settings\SINAL\Escritorio\mech\mech1
11 silly lifecycle mechvibes@2.0.1~start: Args: [ '/d /s /c', 'electron .' ]
12 silly lifecycle mechvibes@2.0.1~start: Returned: code: 1  signal: null
13 info lifecycle mechvibes@2.0.1~start: Failed to exec start script
14 verbose stack Error: mechvibes@2.0.1 start: `electron .`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (K:\Archivos de programa\nodejs\node_modules\npm\lib\utils\lifecycle.js:239:16)
14 verbose stack     at emitTwo (events.js:100:13)
14 verbose stack     at EventEmitter.emit (events.js:185:7)
14 verbose stack     at ChildProcess.<anonymous> (K:\Archivos de programa\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
14 verbose stack     at emitTwo (events.js:100:13)
14 verbose stack     at ChildProcess.emit (events.js:185:7)
14 verbose stack     at maybeClose (internal/child_process.js:850:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
15 verbose pkgid mechvibes@2.0.1
16 verbose cwd K:\Documents and Settings\SINAL\Escritorio\mech\mech1
17 error Windows_NT 5.1.2600
18 error argv "K:\\Archivos de programa\\nodejs\\node.exe" "K:\\Archivos de programa\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
19 error node v5.9.1
20 error npm  v3.7.3
21 error code ELIFECYCLE
22 error mechvibes@2.0.1 start: `electron .`
22 error Exit status 1
23 error Failed at the mechvibes@2.0.1 start script 'electron .'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the mechvibes package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     electron .
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs mechvibes
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls mechvibes
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by hainguyents13** [comment]
``23 error Make sure you have the latest version of node.js and npm installed.``

I think this is the problem, if so, there is nothing I can do

**Comment by andika207** [comment]
@hainguyents13 I have managed to repeat this process on W7 with the latest nodejs 13.8 but I can still see some errors

0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'start'
1 verbose cli ]
2 info using npm@6.13.6
3 info using node@v13.8.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle mechvibes@2.0.1~prestart: mechvibes@2.0.1
6 info lifecycle mechvibes@2.0.1~start: mechvibes@2.0.1
7 verbose lifecycle mechvibes@2.0.1~start: unsafe-perm in lifecycle true
8 verbose lifecycle mechvibes@2.0.1~start: PATH: C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Users\carles\Desktop\mech\mech\node_modules\.bin;C:\Users\carles\AppData\Roaming\npm;C:\Program Files\nodejs\;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Calibre2\;C:\Program Files (x86)\Common Files\Acronis\SnapAPI\;c:\program files\d3dgear;c:\program files\d3dgear;C:\Program Files\nodejs\;C:\Users\carles\AppData\Roaming\npm
9 verbose lifecycle mechvibes@2.0.1~start: CWD: C:\Users\carles\Desktop\mech\mech
10 silly lifecycle mechvibes@2.0.1~start: Args: [ '/d /s /c', 'electron .' ]
11 silly lifecycle mechvibes@2.0.1~start: Returned: code: 1  signal: null
12 info lifecycle mechvibes@2.0.1~start: Failed to exec start script
13 verbose stack Error: mechvibes@2.0.1 start: `electron .`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:321:20)
13 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:321:20)
13 verbose stack     at maybeClose (internal/child_process.js:1026:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
14 verbose pkgid mechvibes@2.0.1
15 verbose cwd C:\Users\carles\Desktop\mech\mech
16 verbose Windows_NT 6.1.7601
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
18 verbose node v13.8.0
19 verbose npm  v6.13.6
20 error code ELIFECYCLE
21 error errno 1
22 error mechvibes@2.0.1 start: `electron .`
22 error Exit status 1
23 error Failed at the mechvibes@2.0.1 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by hainguyents13** [comment]
@andika207 only use nodejs stable version (with even version number like 10.x or 12.x)
try again then let me know the result

**Comment by andika207** [comment]
@hainguyents13 I don't see where it's mentioned the stability of any version...
give me a link of the version you want me to try out

**Comment by hainguyents13** [comment]
Try latest v12 LTS on nodejs home page

**Comment by andika207** [comment]
**12 TLS**

0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'start'
1 verbose cli ]
2 info using npm@6.13.4
3 info using node@v12.15.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle mechvibes@2.0.1~prestart: mechvibes@2.0.1
6 info lifecycle mechvibes@2.0.1~start: mechvibes@2.0.1
7 verbose lifecycle mechvibes@2.0.1~start: unsafe-perm in lifecycle true
8 verbose lifecycle mechvibes@2.0.1~start: PATH: C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Users\carles\Desktop\mech\mech1\node_modules\.bin;C:\Users\carles\AppData\Roaming\npm;C:\Program Files\nodejs\;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Calibre2\;C:\Program Files (x86)\Common Files\Acronis\SnapAPI\;c:\program files\d3dgear;c:\program files\d3dgear;C:\Program Files\nodejs\;C:\Users\carles\AppData\Roaming\npm
9 verbose lifecycle mechvibes@2.0.1~start: CWD: C:\Users\carles\Desktop\mech\mech1
10 silly lifecycle mechvibes@2.0.1~start: Args: [ '/d /s /c', 'electron .' ]
11 silly lifecycle mechvibes@2.0.1~start: Returned: code: 1  signal: null
12 info lifecycle mechvibes@2.0.1~start: Failed to exec start script
13 verbose stack Error: mechvibes@2.0.1 start: `electron .`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:223:5)
13 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:223:5)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
14 verbose pkgid mechvibes@2.0.1
15 verbose cwd C:\Users\carles\Desktop\mech\mech1
16 verbose Windows_NT 6.1.7601
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
18 verbose node v12.15.0
19 verbose npm  v6.13.4
20 error code ELIFECYCLE
21 error errno 1
22 error mechvibes@2.0.1 start: `electron .`
22 error Exit status 1
23 error Failed at the mechvibes@2.0.1 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by andika207** [comment]
**12.15 LTS**

0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'start'
1 verbose cli ]
2 info using npm@6.13.4
3 info using node@v12.15.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle mechvibes@2.0.1~prestart: mechvibes@2.0.1
6 info lifecycle mechvibes@2.0.1~start: mechvibes@2.0.1
7 verbose lifecycle mechvibes@2.0.1~start: unsafe-perm in lifecycle true
8 verbose lifecycle mechvibes@2.0.1~start: PATH: C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Users\carles\Desktop\mech\mech1\node_modules\.bin;C:\Users\carles\AppData\Roaming\npm;C:\Program Files\nodejs\;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Calibre2\;C:\Program Files (x86)\Common Files\Acronis\SnapAPI\;c:\program files\d3dgear;c:\program files\d3dgear;C:\Program Files\nodejs\;C:\Users\carles\AppData\Roaming\npm
9 verbose lifecycle mechvibes@2.0.1~start: CWD: C:\Users\carles\Desktop\mech\mech1
10 silly lifecycle mechvibes@2.0.1~start: Args: [ '/d /s /c', 'electron .' ]
11 silly lifecycle mechvibes@2.0.1~start: Returned: code: 1  signal: null
12 info lifecycle mechvibes@2.0.1~start: Failed to exec start script
13 verbose stack Error: mechvibes@2.0.1 start: `electron .`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:223:5)
13 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:223:5)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
14 verbose pkgid mechvibes@2.0.1
15 verbose cwd C:\Users\carles\Desktop\mech\mech1
16 verbose Windows_NT 6.1.7601
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
18 verbose node v12.15.0
19 verbose npm  v6.13.4
20 error code ELIFECYCLE
21 error errno 1
22 error mechvibes@2.0.1 start: `electron .`
22 error Exit status 1
23 error Failed at the mechvibes@2.0.1 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Comment by hainguyents13** [comment]
I will install windows 7 on a virtual machine and try to install to see what happen. 
Give me some time. 
Do you use windows 7 64 or 32-bit?

**Comment by andika207** [comment]
@hainguyents13 64 bit. 
Are you using W10 then ?
Will you compile a special version with an older nodejs release to fix XP issues ?

**Comment by andika207** [comment]
@hainguyents13 64 bit.
#note: the portable version (Mechvibes-2.0.0-win.zip) doesn't work on W7... I have to install the exe file.
I get the typical error of XP ''*it's not a valid 32 app*''  

Are you using W10 then ?
Will you compile a special version with an older nodejs release to fix XP issues ?

**Comment by hainguyents13** [comment]
@andika207 all current versions is now 64-bit app

Yes I'm using windows 10, I will try  to use older nodejs version to fix your issues, hope it work

**Comment by andika207** [comment]
@hainguyents13 OK thank you.
**Microsoft Telemetry Tools Bundle** this is a repack of tools that might be of your interest for W10

**Comment by hainguyents13** [comment]
> **Microsoft Telemetry Tools Bundle** this is a repack of tools that might be of your interest for W10
It's a security bundle and I dont know what to do with it

**Comment by andika207** [comment]
@hainguyents13 disabling telemetry... you know what I mean.

**Comment by andika207** [comment]
@hainguyents13 disabling telemetry... you know what I mean. 
it's one out the many reasons why I will stick with XP :)

**Comment by andika207** [comment]
@hainguyents13 disabling telemetry... you know what I mean.
it's one out of the many reasons why I will stick with XP :)

**Comment by hainguyents13** [comment]
@andika207 hey but why IE on my Windows XP show "The page cannot be displayed. Cannot find server or DNS error", so I cannot install anything

**Comment by andika207** [comment]
@hainguyents13 You are probably using the default IE6 browser which is very out dated. 
if you need a browser to work on XP pick this one up https://github.com/Feodor2/Centaury

**Comment by hainguyents13** [comment]
@andika207 IE6 cannot access to https links so how can I get the new one?

**Comment by andika207** [comment]
@hainguyents13 IE6 is dead and unusable, download another browser please.
 If you are using a virtual machine you can insert files into it from your W10

**Comment by hainguyents13** [comment]
@andika207 hi, I tried many times but Windows XP is not supported by many packages so I think this cannot be done.
Here are something related: https://github.com/electron/electron/issues/691
Sorry to say this sad thing, windows xp is not supported, and I will close this issue now.
Sorry!

**Comment by andika207** [comment]
@hainguyents13 what packages are you talking about ? 
the last version of noje compatible with XP is **5.9.1**

**Comment by hainguyents13** [comment]
Its electronjs

**Comment by andika207** [comment]
@hainguyents13 this user @weolar said that Electron 5.0 can run with XP

**Comment by hainguyents13** [comment]
@andika207 let me try again with electron 5

**Comment by hainguyents13** [comment]
@andika207 I tried but it still not work, the core package of this app "iohook" does not have prebuilt file for current nodejs version, at least node v5.12.X (https://wilix-team.github.io/iohook/installation.html)

**Comment by andika207** [comment]
@hainguyents13 I didn't know it was so difficult to make the mechavibes app. 
as I stated above the last version compatible with XP is 5.9.1 so the 5.12 is a step ahead. 
try out with the 4.6 instead and sorry for the inconvenience

**Comment by K4sum1** [comment]
Could've been avoided by just not using fucking Electron for an offline desktop app.

**Comment by andika207** [comment]
@K4sum1 well this project seems to be dead in the water. 
I wish someone could fork it without using electron or other compilers that will go EOL sooner than later.

**Comment by andika207** [comment]
> @andika207 @K4sum1 do you know what licence this project uses?

You are asking the wrong persons, for more details get in touch with @NotLazy or @hainguyents13

**Comment by NotLazy** [comment]
> You are asking the wrong persons, for more...

@andika207 @throwaway69420-69420 

There's a license file in the repo now.

**Comment by weolar** [comment]
It is relatively simple for Node.js to support Windows XP. In fact, I have even managed to make Chromium 115 and Electron 22 run on XP. I'm curious to know what technical challenges you have encountered on your end.

**Comment by K4sum1** [comment]
> It is relatively simple for Node.js to support Windows XP. In fact, I have even managed to make Chromium 115 and Electron 22 run on XP. I'm curious to know what technical challenges you have encountered on your end.

How exactly? Is this with OCAPI?

**Comment by K4sum1** [comment]
https://github.com/Skulltrail192/One-Core-API-Binaries

**Comment by K4sum1** [comment]
> It is relatively simple for Node.js to support Windows XP. In fact, I have even managed to make Chromium 115 and Electron 22 run on XP. I'm curious to know what technical challenges you have encountered on your end.

If this is stock XP, I would like to have a copy of this modified Chromium 115 to see if it works.

**Comment by weolar** [comment]
![image](https://github.com/hainguyents13/mechvibes/assets/6235154/3e7a37a3-194f-49d4-b67a-95a56d3c7838)

**Comment by K4sum1** [comment]
I could replicate that screenshot with a VM and unity mode. Unless it's something I can run, I don't have real proof of it existing.

**Comment by weolar** [comment]
Let me think about how to give you the file. The compressed package of Chrome is over 80MB, and it seems that I cannot upload it here on GitHub.

**Comment by K4sum1** [comment]
> Let me think about how to give you the file. The compressed package of Chrome is over 80MB, and it seems that I cannot upload it here on GitHub.

catbox.moe

**Comment by weolar** [comment]
https://t.wss.ink/f/ci6v6exh2js 
try this, see if it can be downloaded, and Please do not spread it

**Comment by weolar** [comment]
This version has just been implemented and there are currently several bugs that have not been fixed

**Comment by K4sum1** [comment]
I tried it under XP x64, and it has the issue where it uses SetProcessDEPPolicy, which doesn't exist on NT 5.2. Same issue that that one Minichrome or whatever had a while ago.

**Comment by weolar** [comment]
What apis are not implemented? I'll replenish them all together

**Comment by K4sum1** [comment]
SetProcessDEPPolicy

**Comment by weolar** [comment]
https://t.wss.ink/f/ci7psiymlxv try it again

**Comment by K4sum1** [comment]
It works now, this is so cool seeing Chromium 115 on XP. Will there be source code available when finished?

**Comment by hainguyents13** [comment]
the Chrome part is done, now the Nodejs, electron and iohook
